x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -195,6 +195,7 @@ public void authorizeClusterAction(
 
     // pkg private for testing
     static boolean checkSameUserPermissions(String action, TransportRequest request, Authentication authentication) {
+        authentication.assertNotRemoteAccess();
```

Comment:
Similarly here, I think it's better to have a allowlist check when receving the request instead of separate checks in downstream code.